agent/agent.go

```diff
@@ -575,8 +586,14 @@ func (a *Agent) flush(
 		}
 
 		select {
+		case <-ctx.Done():
+			logError(a.flushOnce(output, interval, output.Write))
+			return
 		case <-ticker.C:
 			logError(a.flushOnce(output, interval, output.Write))
+		case <-flushRequested:
+			log.Printf("I! [agent] User-requested flush received")
```

Comment:
I wish this was logged only once across all outputs, but I'm not sure it would be worth it.

Reply:
Oh, good point. Alternatively we could not log anything.